File: code/java/ch14/practice/Challenge04MinOpsMakeEqual.java
package ch14.practice;

import java.util.*;

/**
 * Challenge 4: Minimum Operations to Make All Elements Equal
 * Chapter 14: Prefix Sums — The Running Total Trick
 *
 * PROBLEM: Given a sorted array, find min operations (increment/decrement by 1)
 *          to make all elements equal to some existing element.
 *
 * EXAMPLES:
 *   solve([1,2,3]) -> 2
 *   solve([5])     -> 0
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Challenge04MinOpsMakeEqual {
    public static long solve(int[] arr) {
        // TODO: Replace this with your solution
        return 0;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int[] arr = Arrays.stream(sc.nextLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        System.out.println(solve(arr));
        sc.close();
    }
}

File: code/cpp/ch14/practice/challenge_04_min_ops_make_equal.cpp
/*
 * Challenge 4: Minimum Operations to Make All Elements Equal
 * Chapter 14: Prefix Sums — The Running Total Trick
 *
 * PROBLEM: Given a sorted array, find min operations to make all equal.
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */

#include <algorithm>
#include <climits>
#include <iostream>
#include <vector>
using namespace std;

long long solve(vector<int> arr) {
    // TODO: Replace this with your solution
    return 0;
}

// -- Do not change anything below this line --------------------------
int main() {
    int n;
    cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; i++) cin >> arr[i];
    cout << solve(arr) << endl;
    return 0;
}
